src/index.ts

```diff
@@ -191,15 +186,13 @@ export const verify = async function (publicKey: Buffer, msg: Buffer, sig: Buffe
   }
   assert(msg.length > 0, "Message should not be empty");
   assert(msg.length <= 32, "Message is too long");
-  if (ec.verify(msg, sig, publicKey)) {
-    return null;
-  }
+  if (secp256k1.verify(sig, msg, publicKey)) return null;
   throw new Error("Bad signature");
 };
 
-export const derive = async function (privateKeyA: Buffer, publicKeyB: Buffer): Promise<Buffer> {
-  assert(Buffer.isBuffer(privateKeyA), "Bad private key");
-  assert(Buffer.isBuffer(publicKeyB), "Bad public key");
+export const derive = async function (privateKeyA: Uint8Array, publicKeyB: Uint8Array): Promise<Uint8Array> {
+  // assert(Buffer.isBuffer(privateKeyA), "Bad private key");
+  // assert(Buffer.isBuffer(publicKeyB), "Bad public key");
```

Comment:
remove if no longer needed?

what was `isBuffer` doing? do we need to add some checks here?